xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-api/src/main/java/org/xwiki/notifications/internal/page/DefaultPageNotificationDisplayer.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.notifications.internal.page;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.StringUtils;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.notifications.CompositeEvent;
+import org.xwiki.notifications.NotificationDisplayer;
+import org.xwiki.notifications.NotificationException;
+import org.xwiki.notifications.page.PageNotificationEventDescriptor;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.template.Template;
+import org.xwiki.template.TemplateManager;
+import org.xwiki.velocity.VelocityManager;
+
+/**
+ * Implement a {@link NotificationDisplayer} for the event type
+ * {@link org.xwiki.notifications.page.PageNotificationEvent}.
+ *
+ * @version $Id$
+ * @since 9.5RC1
+ */
+@Component
+@Singleton
+@Named(DefaultPageNotificationDisplayer.EVENT_TYPE)
+public class DefaultPageNotificationDisplayer implements NotificationDisplayer
+{
+    /**
+     * Event handled by the displayer.
+     */
+    public static final String EVENT_TYPE = "org.xwiki.notifications.page.PageNotificationEvent";
+
+    private static final String EVENT_BINDING_NAME = "event";
+
+    @Inject
+    private PageNotificationEventDescriptorManager pageNotificationEventDescriptorManager;
+
+    @Inject
+    private TemplateManager templateManager;
+
+    @Inject
+    private VelocityManager velocityManager;
```

Comment:
Sorry, I forgot to remove the manager in this particular displayer ; fixed in 188d8f9.